src/index.js

```diff
@@ -2,14 +2,16 @@ function toVal(mix) {
 	var k, y, str='';
 
 	if (typeof mix === 'string' || typeof mix === 'number') {
-		str += mix;
-	} else if (typeof mix === 'object') {
+		return mix;
```

Comment:
This is a regression. Number type is returned.

Reply:
Added tests for this case. I found this is not regression :-) Because https://github.com/lukeed/clsx/blob/master/src/index.js#L35 converts `toVal` return value to string.